pdns/dnsrecords.cc

```diff
@@ -489,6 +489,227 @@ string EUI64RecordContent::getZoneRepresentation(bool noDot) const
 
 /* EUI64 end */
 
+/* APL start */
+/* https://tools.ietf.org/html/rfc3123 */
+void APLRecordContent::report(void)
+{
+  regist(1, QType::APL, &make, &make, "APL");
+}
+
+// Parse incoming packets (e.g. nsupdate)
+std::shared_ptr<DNSRecordContent> APLRecordContent::make(const DNSRecord &dr, PacketReader& pr) {
+  uint8_t temp;
+  APLRDataElement ard;
+  int processed = 0;
+
+  auto ret=std::make_shared<APLRecordContent>();
+
+  while (processed<dr.d_clen) {
+    pr.xfr16BitInt(ard.d_family);
+    pr.xfr8BitInt(ard.d_prefix);
+    pr.xfr8BitInt(temp);
+    ard.d_n = (temp & 128) >> 7;
+    ard.d_afdlength = temp & 127;
+
+    if (ard.d_family == APL_FAMILY_IPV4) { // IPv4
+      if (ard.d_afdlength > 4) {
+        throw MOADNSException("Invalid IP length for IPv4 APL");
+      }
+      bzero(ard.d_ip4, sizeof(ard.d_ip4));
+      // Call min because we can't trust d_afdlength in an inbound packet
```

Comment:
Do we still need `min()` since the code above should already raise an exception if `d_afdlength` is larger than 4 (and the same for v6 below)?

Reply:
> Note that we prefer the use of `memset()` over `bzero()` for compatibility reasons, as suggested by POSIX.

Thank you for the review Remi. I've committed all requested changes.